lib/src/commands/Commands.ts

```diff
@@ -36,33 +20,39 @@ export class Commands {
     private readonly layoutTreeCrawler: LayoutTreeCrawler,
     private readonly commandsObserver: CommandsObserver,
     private readonly uniqueIdProvider: UniqueIdProvider,
-    private readonly optionsProcessor: OptionsProcessor
+    private readonly optionsProcessor: OptionsProcessor,
+    private readonly layoutProcessor: LayoutProcessor
   ) {}
 
   public setRoot(simpleApi: LayoutRoot) {
     const input = cloneDeep(simpleApi);
-    const root = this.layoutTreeParser.parse(input.root);
+    const processedRoot = this.layoutProcessor.process(input.root, CommandName.SetRoot);
+    const root = this.layoutTreeParser.parse(processedRoot);
 
-    const modals = map(input.modals, (modal) => {
+    const processedModals =
+      input.modals && this.layoutProcessor.process(input.modals, CommandName.SetRoot);
+    const modals = map(processedModals, (modal: any) => {
       return this.layoutTreeParser.parse(modal);
     });
 
-    const overlays = map(input.overlays, (overlay) => {
+    const processedOverlays =
+      input.overlays && this.layoutProcessor.process(input.overlays, CommandName.SetRoot);
+    const overlays = map(processedOverlays, (overlay: any) => {
```

Comment:
Like the previous comment, Overlays should be processed inside the map, then we wouldn't need to check for null or undefined.